### PR TITLE
Introduced decorator to avoid duplicated code

### PR DESCRIPTION
**Pull request by JeffreyDevloo**
Adjusted all tests to use the decorator

### CODE REVIEW

**Comment by JeffreyDevloo** [comment]
Behaviour is a wee bit different now

On fail the tests will collect ovs-workers.service + volumedriver log in their runtime window.
These log gatherings can be finetuned for every test to for instance grep for alba logs instead
However those will be adjusted when needed

Example output on failure:
```
2017-03-30 11:07:06
2017-03-30 11:07:13 85400 +0200 - ovs-node01-1604 - 28229/140616151111424 - scenario/ci_scenario_healthcheck - 5 - ERROR - Healthcheck CI testing failed with error: REMOVE ME AFTER TESTING
2017-03-30 11:07:13 85500 +0200 - ovs-node01-1604 - 28229/140616151111424 - autotests/ci_autotests - 6 - INFO - Finished tests...
Out[3]: 
({'ci.scenarios.healthcheck.healthcheck_run_test': {'blocking': False,
   'case_type': 'FUNCTIONAL',
   'errors': "REMOVE ME AFTER TESTING\n\nLogs collected between 2017-03-30 11:06:36.876233 and 2017-03-30 11:07:06.903984\n\n10.100.199.171 - ovs-workers.service: 2017-03-30 11:07:02 59100 +0200 - ovs-node01-1604 - 12890/140039153186560 - celery/celery.worker.strategy - 6228 - INFO - Received task: ovs.storagerouter.ping[d51468b8-e6ae-466d-bb82-076902fb214c]\n10.100.199.171 - ovs-workers.service: 2017-03-30 11:07:02 59100 +0200 - ovs-node01-1604 - 12890/140039153186560 - celery/celery.worker.autoscale - 6229 - INFO - Scaling up 1 processes.\n10.100.199.171 - ovs-workers.service: 2017-03-30 11:07:02 62300 +0200 - ovs-node01-1604 - 12890/140039153186560 - celery/celery.pool - 6230 - DEBUG - TaskPool: Apply <function _fast_trace_task at 0x7f5d63befcf8> (args:('ovs.storagerouter.ping', 'd51468b8-e6ae-466d-bb82-076902fb214c', ('af4e9c9d-21d2-45b0-9e27-30f874e21c58', 1490864822), {}, {'utc': True, u'is_eager': False, 'chord': None, u'group': None, 'args': ('af4e9c9d-21d2-45b0-9e27-30f874e21c58', 1490864822), 'retries': 0, u'delivery_info': {u'priority': 0, u'redelivered': False, u'routing_key': u'sr.hlU4Wjvvz7a8JrgZ', u'exchange': u'generic'}, 'expires': None, u'hostname': 'celery@ovs-node01-1604', 'task': 'ovs.storagerouter.ping', 'callbacks': None, u'correlation_id': u'd51468b8-e6ae-466d-bb82-076902fb214c', 'errbacks': None, 'timelimit': (None, None), 'taskset': None, 'kwargs': {}, 'eta': None, u'reply_to': u'5d123b96-db42-39bb-8327-3782605a77a5', 'id': 'd51468b8-e6ae-466d-bb82-076902fb214c', u'headers': {}}) kwargs:{})\n10.100.199.171 - ovs-workers.service: 2017-03-30 11:07:02 63400 +0200 - ovs-node01-1604 - 12890/140039153186560 - celery/celery.worker.job - 6231 - DEBUG - Task accepted: ovs.storagerouter.ping[d51468b8-e6ae-466d-bb82-076902fb214c] pid:28261\n10.100.199.171 - ovs-workers.service: 2017-03-30 11:07:02 69900 +0200 - ovs-node01-1604 - 12890/140039153186560 - celery/celery.worker.job - 6232 - INFO - Task ovs.storagerouter.ping[d51468b8-e6ae-466d-bb82-076902fb214c] succeeded in 0.0649337400027s: None\n10.100.199.172 - ovs-workers.service: 2017-03-30 11:07:02 08800 +0200 - ovs-node02-1604 - 7824/140121303815936 - celery/celery.worker.strategy - 6127 - INFO - Received task: ovs.storagerouter.ping[9438f693-214e-45bd-b19b-37828b75cce1]\n10.100.199.172 - ovs-workers.service: 2017-03-30 11:07:02 08800 +0200 - ovs-node02-1604 - 7824/140121303815936 - celery/celery.worker.autoscale - 6128 - INFO - Scaling up 1 processes.\n10.100.199.172 - ovs-workers.service: 2017-03-30 11:07:02 13900 +0200 - ovs-node02-1604 - 7824/140121303815936 - celery/celery.pool - 6129 - DEBUG - TaskPool: Apply <function _fast_trace_task at 0x7f70844e2cf8> (args:('ovs.storagerouter.ping', '9438f693-214e-45bd-b19b-37828b75cce1', ('ecf70553-fec7-43d9-b32c-7d51930153cb', 1490864821), {}, {'utc': True, u'is_eager': False, 'chord': None, u'group': None, 'args': ('ecf70553-fec7-43d9-b32c-7d51930153cb', 1490864821), 'retries': 0, u'delivery_info': {u'priority': 0, u'redelivered': False, u'routing_key': u'sr.qrbEn7sHGQVjdcKh', u'exchange': u'generic'}, 'expires': None, u'hostname': 'celery@ovs-node02-1604', 'task': 'ovs.storagerouter.ping', 'callbacks': None, u'correlation_id': u'9438f693-214e-45bd-b19b-37828b75cce1', 'errbacks': None, 'timelimit': (None, None), 'taskset': None, 'kwargs': {}, 'eta': None, u'reply_to': u'bd083ee0-8312-3617-b29e-dc30b3aa9217', 'id': '9438f693-214e-45bd-b19b-37828b75cce1', u'headers': {}}) kwargs:{})\n10.100.199.172 - ovs-workers.service: 2017-03-30 11:07:02 16200 +0200 - ovs-node02-1604 - 7824/140121303815936 - celery/celery.worker.job - 6130 - DEBUG - Task accepted: ovs.storagerouter.ping[9438f693-214e-45bd-b19b-37828b75cce1] pid:23489\n10.100.199.172 - ovs-workers.service: 2017-03-30 11:07:02 33400 +0200 - ovs-node02-1604 - 7824/140121303815936 - celery/celery.worker.job - 6131 - INFO - Task ovs.storagerouter.ping[9438f693-214e-45bd-b19b-37828b75cce1] succeeded in 0.174609506001s: None\n10.100.199.173 - ovs-workers.service: 2017-03-30 11:07:02 36900 +0200 - ovs-node03-1604 - 6966/140188884285184 - celery/celery.worker.strategy - 6172 - INFO - Received task: ovs.storagerouter.ping[645e37d3-8de1-46d1-809b-ae2e46726def]\n10.100.199.173 - ovs-workers.service: 2017-03-30 11:07:02 36900 +0200 - ovs-node03-1604 - 6966/140188884285184 - celery/celery.worker.autoscale - 6173 - INFO - Scaling up 1 processes.\n10.100.199.173 - ovs-workers.service: 2017-03-30 11:07:02 41200 +0200 - ovs-node03-1604 - 6966/140188884285184 - celery/celery.pool - 6174 - DEBUG - TaskPool: Apply <function _fast_trace_task at 0x7f80406a4cf8> (args:('ovs.storagerouter.ping', '645e37d3-8de1-46d1-809b-ae2e46726def', ('f546c13a-ae4f-4d90-b09a-122363d0b3c7', 1490864822), {}, {'utc': True, u'is_eager': False, 'chord': None, u'group': None, 'args': ('f546c13a-ae4f-4d90-b09a-122363d0b3c7', 1490864822), 'retries': 0, u'delivery_info': {u'priority': 0, u'redelivered': False, u'routing_key': u'sr.htF3Zn5KUWrXZa81', u'exchange': u'generic'}, 'expires': None, u'hostname': 'celery@ovs-node03-1604', 'task': 'ovs.storagerouter.ping', 'callbacks': None, u'correlation_id': u'645e37d3-8de1-46d1-809b-ae2e46726def', 'errbacks': None, 'timelimit': (None, None), 'taskset': None, 'kwargs': {}, 'eta': None, u'reply_to': u'1e3b1d00-a672-30f6-bac5-5a87f96a0754', 'id': '645e37d3-8de1-46d1-809b-ae2e46726def', u'headers': {}}) kwargs:{})\n10.100.199.173 - ovs-workers.service: 2017-03-30 11:07:02 42800 +0200 - ovs-node03-1604 - 6966/140188884285184 - celery/celery.worker.job - 6175 - DEBUG - Task accepted: ovs.storagerouter.ping[645e37d3-8de1-46d1-809b-ae2e46726def] pid:23996\n10.100.199.173 - ovs-workers.service: 2017-03-30 11:07:02 52900 +0200 - ovs-node03-1604 - 6966/140188884285184 - celery/celery.worker.job - 6176 - INFO - Task ovs.storagerouter.ping[645e37d3-8de1-46d1-809b-ae2e46726def] succeeded in 0.101424s: None\n",
   'status': 'FAILED'}},
 None)

```
Example output blocked (unchanged)
```
Out[2]: 
({'ci.scenarios.healthcheck.healthcheck_run_test': {'case_type': 'FUNCTIONAL',
   'errors': None,
   'status': 'BLOCKED'}},
 None)

```
Example output on success (unchanged)
```
Out[2]: 
({'ci.scenarios.healthcheck.healthcheck_run_test': {'case_type': 'FUNCTIONAL',
   'errors': None,
   'status': 'PASSED'}},
 None)

```

File: ci/autotests.py [hunk 1]
```diff
@@ -356,3 +342,127 @@ def _get_description():
...
+        :param since: start collecting logs from this timestamp
+        :type since: str / DateTime
+        :param until: stop collecting when this timestamp is found
+        :type until: str / DateTime
+        :param auto_complete: replaced * with found entries, works around the journalctl flaw
+        :type auto_complete: bool
+        :return: all logs for the components listed
+        :rtype: str
+        """
+        LOGGER.debug('Grepping logs for between {0} and {1}.'.format(since, until))
```
Comment (pploegaert) [inline ci/autotests.py]: Grepping logs between ...

File: ci/autotests.py [hunk 2]
```diff
@@ -356,3 +342,127 @@ def _get_description():
     description_lines.append("{0}".format(_get_package_info()))
 
     return '\n'.join(description_lines)
+
+
+# @todo support 14.04
```
Comment (pploegaert) [inline ci/autotests.py]: This statement can be removed... we won't be doing much for ubuntu 14.04

File: ci/scenarios/api/post_reboot_checks_test-exclude/main.py [hunk 3]
```diff
@@ -16,15 +16,17 @@
...
+from ci.autotests import gather_results
 from ovs.extensions.generic.sshclient import SSHClient
-from ci.api_lib.helpers.storagerouter import StoragerouterHelper
 
 
 class PostRebootChecks(object):
 
     CASE_TYPE = 'AT_QUICK'
-    LOGGER = LogHandler.get(source="scenario", name="ci_scenario_post_reboot_checks")
+    TEST = "ci_scenario_post_reboot_checks"
```
Comment (pploegaert) [inline ci/scenarios/api/post_reboot_checks_test-exclude/main.py]: can you refactor this to TEST_NAME ...?

**Pull request merged**